src/views/Map/CustomerMap.js

```diff
@@ -0,0 +1,221 @@
+    googleScript.async = true;
+    window.document.body.appendChild(googleScript);
+    googleScript.addEventListener('load', this.initAutocomplete);
+  }
+
+  initAutocomplete() {
+    let mapToState = this.handleMapChange;
+    let myLatlng = { lat: 6.60954222424663, lng: 3.370920970651687 };
+    infoWindow = new google.maps.InfoWindow();
+    map = new google.maps.Map(this.googleMapRef.current, {
+      center: myLatlng,
+      zoom: 10,
+      mapTypeId: 'roadmap'
+    });
+
+    map.addListener('click', (mapsMouseEvent) => {
+      // Close the current InfoWindow.
+      infoWindow.close();
+      // Create a new InfoWindow.
+      markers.forEach((marker) => {
+        marker.setMap(null);
+      });
+
+      geoCode(mapsMouseEvent.latLng);
+      console.log(mapsMouseEvent.latLng.toString());
+    });
+
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition((position) => {
+        let pos = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        };
+
+        markers.forEach((marker) => {
+          marker.setMap(null);
+        });
+        markers = [];
+        // markers.push(marker);
+        if (pos.length !== undefined) {
+          geoCode(pos);
+          console.log('navigation working');
+        } else {
+          handleLocationError(true, infoWindow, map.getCenter());
+          console.log(geoCodeValue);
+        }
+        map.setCenter(pos);
+      }, () => {
+        handleLocationError(true, infoWindow, map.getCenter());
+      });
+
+    } else {
+      // Browser doesn't support Geolocation
+      handleLocationError(false, infoWindow, map.getCenter());
+    }
+    function handleLocationError(browserHasGeolocation, infoWindow, pos) {
+      infoWindow.setPosition(pos);
+      infoWindow.setContent(browserHasGeolocation ?
+        'Error: The Geolocation service failed.' :
+        'Error: Your browser doesn\'t support geolocation.');
+      infoWindow.open(map);
+    }
+
+    // Create the search box and link it to the UI element.
+    const input = document.getElementById('pac-input');
+    let searchBox = new google.maps.places.SearchBox(input);
+    map.addListener('bounds_changed', () => {
+      searchBox.setBounds(map.getBounds());
+    });
+
+    markers = [];
+    // Listen for the event fired when the user selects a prediction and retrieve
+    // more details for that place.
+    searchBox.addListener('places_changed', () => {
+      let places = searchBox.getPlaces();
+      if (places.length === 0) {
+        return;
+      }
+      mapToState(places[0].formatted_address);
+      console.log(places[0].formatted_address);
+
+      // Clear out the old markers.
+      markers.forEach((marker) => {
+        marker.setMap(null);
+      });
+      markers = [];
+
+      // For each place, get the icon, name and location.
+      let bounds = new google.maps.LatLngBounds();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['bounds' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/BuildForSDG/Team-064-Frontend/pullRequest?prid=5703258)